pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1082,6 +1083,45 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 				}
 			},
 		},
+		{
+			name: "setSyncSetFailedCondition should be present",
+			existing: []runtime.Object{
+				testInstalledClusterDeployment(time.Now()),
+				createSyncSetInstanceObj(hivev1.ApplyFailureSyncCondition),
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cd := getCD(c)
```

Comment:
might want to assert that we didn't get back a nil clusterDeployment from getCD()